main.py

```diff
@@ -28,13 +31,27 @@ def __init__(self):
 class KeywordQueryEventListener(EventListener):
 
     def on_event(self, event, extension):
-        query = r"""SELECT
-            em.name, em.code, em.icon, em.keywords,
-            skt.code AS skt_code, skt.icon AS skt_icon
-            FROM emoji AS em
-            LEFT JOIN skin_tone AS skt ON skt.name = em.name AND tone = ?
-            WHERE name_search LIKE ?
-            LIMIT 8"""
+        search_limit = extension.preferences['search_limit']
+
+        try:
+            search_limit = search_limit.strip()
+            search_limit = int(search_limit)
+
+            if search_limit < SEARCH_LIMIT_MIN:
+                search_limit = SEARCH_LIMIT_MIN
+            elif search_limit > SEARCH_LIMIT_MAX:
+                search_limit = SEARCH_LIMIT_MAX
+        except Exception as e:
+            search_limit = SEARCH_LIMIT_DEFAULT
+
+        query = 'SELECT \
+            em.name, em.code, em.icon, em.keywords, \
+            skt.code AS skt_code, skt.icon AS skt_icon \
+            FROM emoji AS em \
+            LEFT JOIN skin_tone AS skt ON skt.name = em.name AND tone = ? \
+            WHERE name_search LIKE ? \
+            LIMIT {}'.format(search_limit)
```

Comment:
Thanks for the PR @therj 

I like the improvement you have submitted.

I have only one comment about the code: it's considered best practice to pass any variables in SQL query using argument placeholders (`?`) and then the values themselves you can pass in `conn.execute` down below.
Could you make that change please?

Please also note that there's a merge conflict with master.

Thanks!